hazelcast/src/main/java/com/hazelcast/internal/memory/impl/AlignmentAwareMemoryAccessor.java

```diff
@@ -0,0 +1,1151 @@
+            return (short) ((byte0 << 8) + byte1);
+        }
+
+        @Override
+        public short getShortVolatile(Object o, long offset) {
+            int byte1 = UNSAFE.getByteVolatile(o, offset) & 0xFF;
+            int byte0 = UNSAFE.getByteVolatile(o, offset + 1) & 0xFF;
+            return (short) ((byte0 << 8) + byte1);
+        }
+
+        @Override
+        public int getInt(long address) {
+            int byte3 = (UNSAFE.getByte(address) & 0xFF);
+            int byte2 = (UNSAFE.getByte(address + 1) & 0xFF) << 8;
+            int byte1 = (UNSAFE.getByte(address + 2) & 0xFF) << 16;
+            int byte0 = (UNSAFE.getByte(address + 3) & 0xFF) << 24;
+            return byte3 + byte2 + byte1 + byte0;
+        }
+
+        @Override
+        public int getInt(Object o, long offset) {
+            int byte3 = (UNSAFE.getByte(o, offset) & 0xFF);
+            int byte2 = (UNSAFE.getByte(o, offset + 1) & 0xFF) << 8;
+            int byte1 = (UNSAFE.getByte(o, offset + 2) & 0xFF) << 16;
+            int byte0 = (UNSAFE.getByte(o, offset + 3) & 0xFF) << 24;
+            return byte3 + byte2 + byte1 + byte0;
+        }
+
+        @Override
+        public int getIntVolatile(Object o, long offset) {
+            int byte3 = (UNSAFE.getByteVolatile(o, offset) & 0xFF);
+            int byte2 = (UNSAFE.getByteVolatile(o, offset + 1) & 0xFF) << 8;
+            int byte1 = (UNSAFE.getByteVolatile(o, offset + 2) & 0xFF) << 16;
+            int byte0 = (UNSAFE.getByteVolatile(o, offset + 3) & 0xFF) << 24;
+            return byte3 + byte2 + byte1 + byte0;
+        }
+
+        @Override
+        public float getFloat(long address) {
+            return Float.intBitsToFloat(getInt(address));
+        }
+
+        @Override
+        public float getFloat(Object o, long offset) {
+            return Float.intBitsToFloat(getInt(o, offset));
+        }
+
+        @Override
+        public float getFloatVolatile(Object o, long offset) {
+            return Float.intBitsToFloat(getIntVolatile(o, offset));
+        }
+
+        @Override
+        public long getLong(long address) {
+            long byte7 = (long) (UNSAFE.getByte(address) & 0xFF);
+            long byte6 = (long) (UNSAFE.getByte(address + 1) & 0xFF) << 8;
+            long byte5 = (long) (UNSAFE.getByte(address + 2) & 0xFF) << 16;
+            long byte4 = (long) (UNSAFE.getByte(address + 3) & 0xFF) << 24;
+            long byte3 = (long) (UNSAFE.getByte(address + 4) & 0xFF) << 32;
+            long byte2 = (long) (UNSAFE.getByte(address + 5) & 0xFF) << 40;
+            long byte1 = (long) (UNSAFE.getByte(address + 6) & 0xFF) << 48;
+            long byte0 = (long) (UNSAFE.getByte(address + 7) & 0xFF) << 56;
+            return byte7 + byte6 + byte5 + byte4 + byte3 + byte2 + byte1 + byte0;
+        }
+
+        @Override
+        public long getLong(Object o, long offset) {
+            long byte7 = (long) (UNSAFE.getByte(o, offset) & 0xFF);
```

Comment:
In this implementation, no way, I have not added. It uses underlying platform's endian format